ls1intum/Artemis
# `Exam mode`: Improve exam import validation

### Checklist
#### General


#### Server
#### Client
#### Changes affecting Programming Exercises

### Motivation and Context


When importing an exam or an exercise group it is possible to input the same short name and/or title for multiple programming exercises. This then leades to issues when creating a project on VSC/CI. In this PR a check was added making sure that titles and short names of all programming exercises being imported are unique.

### Description

When the same short name or title is inputed for multiple programming exercise, an alert is sent informing the instructor of this.  The effected exercise's short name / title is then removed and has to be set again. In case there are no duplicated short names and titles, it is then checked if a project with the same key and name already exists on the VSC/CI as before.

### Steps for Testing

Prerequisites:
- 1 Instructor
- 1 Exam with >1 programming exercise to be imported
- 1 Exercise Group with >1 programming exercise to be imported

1. Go to Course Management and select a course
2. Click on Exams and there the Import an exam button
4. Select an exam with at least 2 programming exercise
5. Use the same title / short name for multiple programming exercises and click on import
6. An Alert should appear and the corresponding duplicated field should now be empty and marked with red
7. Go to Exercise Groups and select Import Exercise Group
8. Select an an exercise group with at least 2 programming exercises
9. Again set the same short name, title and import
10. Alert should appear and the duplicated field should now be empty

### Review Progress






#### Performance Review
#### Code Review
#### Manual Tests

### Test Coverage






same as develop

### Screenshots


After clicking imput, the duplicated title is removed and has to be set again:
![image](https://github.com/ls1intum/Artemis/assets/73339358/c45af662-06f0-4f4a-a88d-64ee9a44f42e)


https://github.com/ls1intum/Artemis/assets/73339358/5bb45d7a-f574-4b08-b6ce-0de0b66e2e1e
